### PR TITLE
File actions for Storage

### PR DESCRIPTION
**Pull request by alex-deaconu**
Get File URL
Move to Trash, Restore from Trash, Delete from Trash
Download files & folders

@ezequielc FYI. Thanks.

### CODE REVIEW

File: web/scripts/storage/controllers/ctr-file-actions.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+          filesFactory.filesDetails.checkedItemsCount !== 1;
+      };
+
+      $scope.isDisabledDownloadButton = $scope.isDisabledTrashButton =
+        $scope.isDisabledRestoreButton = $scope.isDisabledDeleteButton =
+        function () {
+          return filesFactory.filesDetails.checkedItemsCount < 1;
+        };
+
+      angular.element('#deleteForm').submit(function (event) {
```
Comment (ezequielc) [inline web/scripts/storage/controllers/ctr-file-actions.js]: @alex-deaconu Do you know where/when this is used? I removed this code in my branch as it does not seem to be in use anymore. Thanks

Comment (alex-deaconu) [inline web/scripts/storage/controllers/ctr-file-actions.js]: @ezequielc yes, here: https://github.com/Rise-Vision/storage-client/blob/master/web/partials/delete-modal.html#L1

This is supposed to allow the user to 'submit' the form by pressing enter. However, in our new implementation we use the `confirmInstance` component.

Maybe we should add a card to duplicate this functionality in `confirmInstance`.

Thanks.

File: web/scripts/storage/services/svc-download.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+          svc.downloadFiles(files, bucketName, 1000);
+        }, delay, false);
+      };
+
+      svc.downloadFile = function (file) {
+        storage.getSignedDownloadURI(file).then(
+          function (resp) {
+            if (resp.result) {
+              var downloadName = file.name.replace('--TRASH--/', '');
```
Comment (ezequielc) [inline web/scripts/storage/services/svc-download.js]: @alex-deaconu As a future refactor, I think we should create a constant for `--TRASH--/`. We are using it in many places. Cheers

Comment (alex-deaconu) [inline web/scripts/storage/services/svc-download.js]: @ezequielc definitely!

File: web/scripts/storage/controllers/ctr-file-actions.js (hunk 1 above)
Comment (ezequielc) [inline web/scripts/storage/controllers/ctr-file-actions.js]: @alex-deaconu Yes, that could be done. Thanks

**Comment by ezequielc** [comment]
@alex-deaconu Code looks good. As it is a work in progress, there are some issues/improvements that we are going to address in the next PRs. Thanks

**Comment by alex-deaconu** [comment]
@ezequielc squashed & ready for merge. Thanks.

**Pull request merged**